src/types.ts

```diff
@@ -1627,18 +1627,37 @@ export enum NFTOwnershipWalletType {
     "END_USER_WALLET" = "END_USER_WALLET",
 }
 
+export enum ContractTemplateType {
+    FUNGIBLE_TOKEN = "FUNGIBLE_TOKEN",
+    NON_FUNGIBLE_TOKEN = "NON_FUNGIBLE_TOKEN",
+    NON_TOKEN = "NON_TOKEN",
+    UUPS_PROXY = "UUPS_PROXY",
+}
+
+export type SupportedContractTemplateType = ContractTemplateType.FUNGIBLE_TOKEN | ContractTemplateType.NON_FUNGIBLE_TOKEN;
+
+export enum ContractInitializationPhase {
+    ON_DEPLOYMENT = "ON_DEPLOYMENT",
+    POST_DEPLOYMENT = "POST_DEPLOYMENT",
+}
+
 export interface ContractUploadRequest {
     name: string;
     description: string;
     longDescription: string;
     bytecode: string;
     sourcecode: string;
+    type?: ContractTemplateType;
+    implementationContractId?: string;
+    initializationPhase: ContractInitializationPhase;
     compilerOutputMetadata?: object;
+    inputFieldsMetadata?: object;
```

Comment:
we have a DTO for this...

Reply:
`InputFieldsMetadata` interface was added